src/protocol/logpoint.ts

```diff
@@ -274,7 +274,8 @@ async function setMultiSourceLogpoint(
     }
   };
 
-  if (condition || (showInConsole && !primitives)) {
+  const shouldGetResults = condition || (showInConsole && !primitives);
```

Comment:
This is not your fault, but I'm struggling with these names. I assume that `condition` means the user-specified `if` statement, `showInConsole` means that we're expected to show the results of this in the console (when is this not true?) and I'm not sure what primitives is? And `shouldGetResults` is kind of confusing because I would expect that if `shouldGetResults` is `false` we wouldn't call `runAnalysis`. So I guess I'm not sure what getting results means here!  Anyways, the change looks fine I think, or at least, I believe you that it does what it says on the tin, but it might be a nice time to revisit some of this naming if you have all of the context in your head!

Reply:
Yeah it takes a second:
- `showInConsole`: logpoints (or print statements) show up in the console. But we also happen to use it as a roundabout way to get the number of hits for a line. It's still an analysis, but we just care about the hits count. And the way we indicate that we're only trying to get the hits is with `showInConsole` set to `false`.
- `shouldGetResults`: Similar-ish to above. If we just need the hits, we don't need the backend to actually run the full analysis.
- `primitives` is a flag for checking if the entire expression is all primitive (`"1", 2, false`), in which case we don't need to run the full analysis since there's nothing specific to each pause point to evaluate.

Reply:
Gonna merge it but I totally understand how this is all sorts of weird as it's currently written. I'm adding it to my try-cleaning-it-up-on-cleanup-friday-afternoon list!